bugs/PENDING_SUBMIT/0392-BAD-IMAGE-1e394f58-opt-O-test-host-Nvidia/reduced_1_result/test.amber

```diff
@@ -0,0 +1,324 @@
+         %22 = OpVariable %21 Input
+         %27 = OpTypeImage %10 2D 0 0 0 1 Unknown
+         %28 = OpTypeSampledImage %27
+         %29 = OpTypePointer UniformConstant %28
+         %30 = OpVariable %29 UniformConstant
+         %55 = OpTypeBool
+         %58 = OpConstant %6 32
+         %72 = OpConstant %6 1
+         %74 = OpTypePointer Output %20
+         %75 = OpVariable %74 Output
+         %78 = OpConstant %10 0.03125
+         %82 = OpConstant %10 1
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+         %23 = OpLoad %20 %22
+         %24 = OpVectorShuffle %11 %23 %23 0 1
+         %25 = OpVectorTimesScalar %11 %24 %14
+         %31 = OpLoad %28 %30
+         %33 = OpImageSampleImplicitLod %20 %31 %25
+         %36 = OpCompositeExtract %10 %33 1
+         %39 = OpFSub %11 %25 %15
+         %42 = OpFAdd %10 %36 %18
+         %46 = OpImageSampleImplicitLod %20 %31 %39
+         %47 = OpCompositeExtract %10 %46 1
+               OpBranch %48
+         %48 = OpLabel
+         %90 = OpPhi %11 %39 %5 %63 %49
+         %89 = OpPhi %6 %9 %5 %73 %49
+         %88 = OpPhi %10 %42 %5 %66 %49
+         %87 = OpPhi %10 %47 %5 %70 %49
+         %56 = OpFOrdLessThan %55 %87 %88
+         %59 = OpSLessThan %55 %89 %58
+         %60 = OpLogicalAnd %55 %56 %59
+               OpLoopMerge %50 %49 None
+               OpBranchConditional %60 %49 %50
+         %49 = OpLabel
+         %63 = OpFSub %11 %90 %15
+         %66 = OpFAdd %10 %88 %18
+         %69 = OpImageSampleImplicitLod %20 %31 %63
+         %70 = OpCompositeExtract %10 %69 1
+         %73 = OpIAdd %6 %89 %72
+               OpBranch %48
+         %50 = OpLabel
+         %77 = OpConvertSToF %10 %89
+         %79 = OpFMul %10 %77 %78
+         %86 = OpCompositeConstruct %20 %79 %79 %79 %82
+               OpStore %75 %86
+               OpReturn
+               OpFunctionEnd
+END
+
+# uniforms for reference
+
+# tex
+SAMPLER reference_tex
+
+
+BUFFER reference_framebuffer FORMAT B8G8R8A8_UNORM
+
+PIPELINE graphics reference_pipeline
+  ATTACH reference_vertex_shader
+  ATTACH reference_fragment_shader
+  FRAMEBUFFER_SIZE 256 256
+  BIND BUFFER reference_framebuffer AS color LOCATION 0
+  BIND BUFFER default_texture AS combined_image_sampler SAMPLER reference_tex DESCRIPTOR_SET 0 BINDING 0
+END
+CLEAR_COLOR reference_pipeline 0 0 0 255
+
+CLEAR reference_pipeline
+RUN reference_pipeline DRAW_RECT POS 0 0 SIZE 256 256
+
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+#
+# precision highp int;
+#
+# precision highp float;
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# layout(set = 0, binding = 0) uniform sampler2D tex;
+#
+# void main()
+# {
+#  int i = 0;
+#  vec2 uvstep = vec2(1.0) * (1.0 / 256.0);
+#  float slope = 2.0 / 256.0;
+#  vec2 coord = gl_FragCoord.xy * (1.0 / 256.0);
+#  float refh = ((vec4(mat3(texture(tex, coord), 1.0, 1.0, 0.0, 0.0, 0.0))) / vec4(1.0, 1.0, 1.0, 1.0)).y;
```

Comment:
As constructing a vec4 from a mat3 is unusual, it would be worth having a comment above this line to explain what's going on - ideally this would spell out what the mat3 looks like, and why the vec4 obtained from the mat3 is exactly texture(tex, coord).

Reply:
Fixed. Showing what mat3 would look like exactly would be a bit difficult, since the texture is very detailed. But I've written the comment in a way that it hopefully explains it a bit. Hopefully I've also managed to explain in the same comment why the vec4 obtained from the mat3 is exactly texture(...).

The changes are in the reduced_1/ directory instead of the reduced_1_result/.